packages/plugin-core/src/test/suite-integ/NoteLookupCommand.test.ts

```diff
@@ -671,5 +713,611 @@ suite("NoteLookupCommand", function () {
         }
       });
     });
+
+    test("horizontal split basic", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
+        },
+        onInit: async ({ vaults, engine }) => {
+          const cmd = new NoteLookupCommand();
+          stubVaultPick(vaults);
+
+          // close all editors before running.
+          VSCodeUtils.closeAllEditors();
+
+          await VSCodeUtils.openNote(engine.notes["foo"]);
+          await cmd.run({
+            initialValue: "bar",
+            splitType: "horizontal",
+            noConfirm: true,
+          });
+          const barEditor = VSCodeUtils.getActiveTextEditor();
+          expect(barEditor!.viewColumn).toEqual(2);
+
+          await cmd.run({
+            initialValue: "foo.ch1",
+            splitType: "horizontal",
+            noConfirm: true,
+          })
+          const fooChildEditor = VSCodeUtils.getActiveTextEditor();
+          expect(fooChildEditor!.viewColumn).toEqual(3);
+
+          done();
+        }
+      });
+    });
+
+    test("horizontal split modifier toggle", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
+        },
+        onInit: async ({ vaults }) => {
+          const cmd = new NoteLookupCommand();
+          stubVaultPick(vaults);
+
+          const { controller } = await cmd.gatherInputs({
+            splitType: "horizontal"
+          });
+
+          controller.quickpick.show();
+
+          let { horizontalSplitBtn } = getSplitTypeButtons(
+            controller.quickpick.buttons
+          );
+
+          expect(horizontalSplitBtn?.pressed).toBeTruthy();
+
+          await controller.onTriggerButton(horizontalSplitBtn);
+
+          ({ horizontalSplitBtn } = getSplitTypeButtons(
+            controller.quickpick.buttons
+          ));
+
+          expect(horizontalSplitBtn.pressed).toBeFalsy();
+
+          done();
+        }
+      });
+    });
+
+    test("copyNoteLink basic", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
+        },
+        onInit: async ({ vaults, }) => {
+          const cmd = new NoteLookupCommand();
+          stubVaultPick(vaults);
+
+          const { controller } = await cmd.gatherInputs({
+            initialValue: "foo"
+          });
+          controller.quickpick.show();
+
+          const { copyNoteLinkBtn } = getEffectTypeButtons(
+            controller.quickpick.buttons
+          );
+
+          // hack: need to wait here a bit.
+          // TODO: A more elegant way to do this.
+          await new Promise(resolve => setTimeout(resolve, 500));
```

Comment:
example of test that needs waiting @kevinslin

Reply:
copying should only happen after `onTriggerButton`. shouldn't await come after that? also removed it locally and this worked fine

Reply:
For some reason `await clipboard.readText()` kept grabbing whatever was in my clipboard before, not the note links. Maybe it's not a flaky test and it just doesn't work properly 😅  I'll look into it.

Reply:
It's really weird because it worked last night and now it doesn't regardless of the `setTimeout`.